extensions/2.0/Vendor/EXT_mesh_primitive_edge_visibility/README.md

```diff
@@ -0,0 +1,221 @@
+
+The extension's `lineStrings` property provides an alternate representation for some or all of the hard edges that would otherwise be encoded in `visibility`. It is an array in which each entry is an object with the following properties:
+- `indices` (required): the index of an accessor of type `SCALAR` and component type `UNSIGNED_BYTE` (5121), `UNSIGNED_SHORT` (5123), or `UNSIGNED_INT` (5125) encoding one or more line strings as indices into the triangle mesh's vertex attribute array. Each line string is encoded as a series of two or more indices. Multiple line strings can be encoded by inserting a "primitive restart" value (that is, the maximal possible value for the accessor's component type, like 255 for `UNSIGNED_BYTE`) between each line string. The `indices` buffer **MUST NOT** begin or end with a primitive restart value, and **MUST NOT** contain consecutive primitive restart values.
+- `material` (optional): the index of the material with which the edges encoded by `indices` are to be drawn. If omitted, the extension's own `material` property is used, if defined; otherwise, the material of the triangle mesh primitive is used.
+
+A hard edge encoded in `lineStrings` **MUST NOT** also be encoded in `visibility`.
+
+The `lineStrings` property **MUST** be defined if `visibility` is not defined - otherwise, the extension would encode no edges.
+
+#### Examples
+
+Consider the following simple triangle fan with triangles described by the index list `[0,1,2, 0,2,3, 0,3,4, 0,4,5]`:
+```
+  2______3______4
+  |＼    |    ／|
+  |  ＼  |  ／  |
+  |____＼|／____|
+  1      0      5
+```
+
+Assume that the top and bottom edges are hard edges, to be encoded as a pair of line strings with indices `[2,3,4]` and `[1,0,5]`, respectively; and the vertical edges are hard edges, to be encoded as simple line segments. Then the `visibility` bitfield should encode only the vertical edges, with visibility values `[0,2,0, 0,0,2, 3,0,0, 0,2,0]`.
+
+##### Single material
+
+Assume that the top and bottom edges are to be drawn using the same material, different from the material used to draw the vertical edges, and the indices are encoded as unsigned bytes. Then the index buffer would look like `[2,3,4, 255, 1,0,5]`, where `255` is the primitive restart value separating the two line strings. The `lineStrings` property would look something like this:
+```
+{
+    "lineStrings": [
+        {
+            "indices": 5,
+            "material": 1
+        }
+    ]
+}
+```
+
+##### Two materials
+
+Assume that the top edge is to be drawn using the same material as the vertical edges, and the bottom edge is to be drawn using a different material. Then there would be two index buffers - `[2,3,4]` and `[1,0,5]` - and the `lineStrings` property would look something like this:
```

Comment:
In this case, why wouldn't the top edge just be included together with the vertical edges in the visibility indices like `[0,2,0, 0,2,2, 3,2,0, 0,2,0]`? Doesn't that produce the same visual effect?